src/data/polynomial/ring_division.lean

```diff
@@ -160,6 +160,52 @@ begin
   rw [← nat_degree_mul hp₁ hq₂, ← nat_degree_mul hp₂ hq₁, h_eq]
 end
 
+variables [char_zero R]
+
+@[simp]
+lemma degree_bit0_eq (p : R[X]) : degree (bit0 p) = degree p :=
+begin
+  rw [bit0_eq_two_mul, degree_mul, (_ : (2 : polynomial R).degree = 0), zero_add],
+  rw (by simp : (2 : polynomial R) = (C 2)),
+  exact polynomial.degree_C two_ne_zero',
+end
+
+@[simp]
+lemma nat_degree_bit0_eq (p : R[X]) : nat_degree (bit0 p) = nat_degree p :=
```

Comment:
I would prefer if we removed the `eq` suffix from these lemmas, but it seems like this leads to a name clash. Perhaps the current lemmas can be suffixed with `le`, and the docs can be made to cross-reference one another?

Reply:
Personally I don't mind it this way, as these lemmas have the more specific assumptions, and so apply less generally.
I would also be fine if we did as you suggest and swap them though. I definitely agree the docs should cross-reference.
Perhaps @adomani has an opinion?

Reply:
I do not have a strong opinion either way.  I think that the convention would be that `_eq` refers to a lemma of the form `nat_degree (bit0 P) = n -> ...`, but I am not sure that this is consistently used (nor an actual convention).

In general, I find all the lemma names about `(nat_)degree`s very long, but they are also fairly clear and easy to use.  A verbose alternative would be `nat_degree_bit0_eq_nat_degree`: less ambiguous, but longer.

The current version is good, though!

Reply:
Going back to this: I actually think renaming the current `nat_degree_bit0/1` lemmas to become `nat_degree_bit0/1_le` is probably a good idea.  I might have been the one introducing those lemmas in the first place. :shrug: